### ISSUE TITLE
Support passing license_key into newrelic.agent.initialize

### ISSUE THREAD

**Issue by a-recknagel** [newrelic/newrelic-python-agent#28]
### Is your feature request related to a problem? Please describe.
I'd like to have the option to pass the license key directly into the agent initialization function. I don't want to include it in the newrelic.ini file, since the key is a secret and is only available to the application during deployment.

### Feature Description
```python
if my_app.config["environment"] in ("stage", "live"):
    newrelic.agent.initialize(
        config_file="newrelic.ini",
        environment=my_app.config["environment"],
        log_level=my_app.config["loglevel"],
        license_key=my_app.config["new_relic_license_key"],  # <-- this is what would it look like
    )
    newrelic.agent.register_application()
```

### Describe Alternatives
I could also have newrelic read the license key from the env, but our app doesn't use the env for configuration as of yet. And splitting configuration options over multiple avenues would be a bad design, so we'd rather not do it.

As a workaround, we currently manipulate `os.environ` before importing any newrelic modules.

### Additional Context
I tried to write a PR myself, but the config parsing for newrelic is _wild_, and I couldn't get it to work by myself. 

### Priority
[Nice to Have]

**Comment by a-feld** [comment]
Hi @a-recknagel !

I'd love to understand more about your use case. How are your app configurations loaded and how are your secrets handled? Are you injecting a config file with secrets at build time? Or are the variables in the config encrypted with a key available in each environment? Something else?

It's not usually recommended to import configurations prior to loading the agent, as doing so can cause modules that the agent monkeypatches to be imported prior to setting up instrumentation hooks, resulting in undefined agent behavior.

That being said, there is another alternative. The agent has an API to change the license key (and any other setting) programatically, prior to registration:

```python
if my_app.config["environment"] in ("stage", "live"):
    newrelic.agent.initialize(
        config_file="newrelic.ini",
        environment=my_app.config["environment"],
        log_level=my_app.config["loglevel"],
    )
    newrelic.agent.global_settings().license_key = my_app.config["new_relic_license_key"]
    newrelic.agent.register_application()
```

The existing `initialize` arguments are only used to control how the config file is read, which is why there's not additional arguments on that function.
* config_file = the file to load
* environment = the section of the config file to load (in addition to the main section)
* ignore_errors = whether errors are ignored during loading of the configuration
* log_file = the log file that is used during the first part of configuration load - this is overridden by a log_file entry in the config file if one is specified
* log_level =  the log level that is used during the first part of configuration load - this is overridden by a log_level entry in the config file if one is specified

The arguments other than `config_file` and `environment` are not commonly used since `ignore_errors` defaults to `True` and `log_file`/`log_level` are generally specified in the configuration file (overriding the arguments supplied to initialize).

**Comment by a-recknagel** [comment]
> Are you injecting a config file with secrets at build time?

No, our build step just bundles our code with a default config into a wheel and throws it plus all its dependencies into a docker image. The environment-specific config gets injected when we the container starts up, and the app then delegates tool-related settings and secrets by hand into them.I don't know if that makes a difference though.

But I think I can capture the part that we didn't want our orchestration to do in this before/after:

before adding newrelic integration:
```bash
$ docker run -p 5000 -m conf.ini:conf.ini my_app 
```
after newrelic integration, and before we found the `os.environ` workaround
```bash
$ docker run -p 5000 -m conf.ini:conf.ini -e NEW_RELIC_LICENSE_KEY my_app 
```
Thanks for the tip with `agent.global_settings()` by the way, that's a lot better.

---
> The existing initialize arguments are only used to control how the config file is read, which is why there's not additional arguments on that function.

I guess I misunderstood its intention then. If ini-file and env-variables are the way to go about configuring newrelic, then my point might be moot.

Looking back, I saw the keywords on `agent.initialize` and thought "Surely, with the license being the only non-optional setting we should be able to provide it up-front here", but I guess that option is only given for log-related settings so that it is possible to log configuration errors.

**Comment by a-recknagel** [comment]
@a-feld I tried replacing `os.environ["NEW_RELIC_LICENSE_KEY"] = my_app.config["new_relic_license_key"]` with `newrelic.agent.global_settings().license_key = my_app.config["new_relic_license_key"]`, and after the server starts up newrelic panics with

```
2020-10-26 19:46:58,112 [CRITICAL] newrelic.core.agent_protocol: Disconnection of the agent has been requested by the data collector for the application where the agent run was None. Please contact New Relic support for further information.
2020-10-26 19:46:58,113 [ERROR] newrelic.core.application: The New Relic service has requested that the agent stop attempting to connect. The agent will no longer attempt a connection with New Relic. Your application must be manually restarted in order to connect to New Relic.
```
This is the same kind of issue I ran into while trying to come up with a PR for the feature.

**Comment by a-feld** [comment]
That can happen if the agent registers before updating the license key.  If you're able to dump debug logs, we should see where the agent is attempting registration. The update to the license key needs to happen before that point.

Order must be this:
```python
# Set license key
newrelic.agent.global_settings().license_key = my_app.config["new_relic_license_key"]

# (optional, but must occur after the license key is set)
newrelic.agent.register_application()
```

**Comment by a-recknagel** [comment]
Yes, I have set it up in that way. Unless I have messed something up in a very strange way, you should also be able to reproduce it with this gist: https://gist.github.com/a-recknagel/188330ad877b310f883e3d9bf4cd8379, given an actually valid license key.

**Comment by a-feld** [comment]
I ran that gist and everything worked ok for me. Can you report the output of this script? (with a valid license key)

```
import newrelic.agent
newrelic.agent.global_settings().log_file = "stderr"
newrelic.agent.global_settings().log_level = "INFO"

newrelic.agent.initialize("newrelic.ini")
newrelic.agent.global_settings().license_key = "YOUR_LICENSE_KEY_HERE"
newrelic.agent.register_application(timeout=10.0)
```

**Comment by a-recknagel** [comment]
```
$ uvicorn app:app
DEBUG:newrelic.config:agent configuration file was newrelic.ini
2020-10-26 20:40:32,144 (25002/MainThread) newrelic.core.agent INFO - New Relic Python Agent (5.22.0.151)
INFO:newrelic.core.agent:New Relic Python Agent (5.22.0.151)
2020-10-26 20:40:32,514 (25002/NR-Activate-Session/HelloWorld) newrelic.core.agent_protocol CRITICAL - Disconnection of the agent has been requested by the data collector for the application where the agent run was None. Please contact New Relic support for further information.
CRITICAL:newrelic.core.agent_protocol:Disconnection of the agent has been requested by the data collector for the application where the agent run was None. Please contact New Relic support for further information.
2020-10-26 20:40:32,514 (25002/NR-Activate-Session/HelloWorld) newrelic.core.application ERROR - The New Relic service has requested that the agent stop attempting to connect. The agent will no longer attempt a connection with New Relic. Your application must be manually restarted in order to connect to New Relic.
ERROR:newrelic.core.application:The New Relic service has requested that the agent stop attempting to connect. The agent will no longer attempt a connection with New Relic. Your application must be manually restarted in order to connect to New Relic.
INFO:     Started server process [25002]
INFO:uvicorn.error:Started server process [25002]
INFO:     Waiting for application startup.
INFO:uvicorn.error:Waiting for application startup.
INFO:     Application startup complete.
INFO:uvicorn.error:Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
INFO:uvicorn.error:Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
```

If I plug in the key with os.environ, it works. And I made extra sure there is no typo in the key.

**Comment by a-feld** [comment]
OK can you set up logging at the debug level then? I can't reproduce the problem on my end unfortunately 😞

**Comment by a-feld** [comment]
OH! You're in the EU! Ha!! OK. You'll also need to set the host!
```python
from newrelic.core.config import default_host
newrelic.agent.global_settings().host = default_host("YOUR_KEY_HERE")
```

**Comment by a-recknagel** [comment]
It's working now, thank you very much for your patience in this debugging session!

**Comment by a-feld** [comment]
Yeah thanks for bringing this up! We should probably consider API usage when using an EU license key. This is something we'll chat about on the team!
